### DISCUSSION TITLE
Docker Home Assistant Gazpar custom component installation

### DISCUSSION THREAD

**Discussion by pbranly** [ssenart/PyGazpar#8] [General]
Bonjour,
J’ai finalement réussi à faire fonctionner le programme après installation.
Pour y arriver je fais:
cd /home/PyGazpar. < ——- répertoire d’installation
source venv/bin/activate < —— visiblement nécessaire sinon on n’entre pas dans l’environnement virtuel

pygazpar -u 'xxxxxx@xxxx .fr' -p 'Trucmuche' -w '/usr/local/bin/geckodriver' -s 30 -t /home/PyGazpar
A la fin du process, j’ai bien l’affichage de toutes les consommations dans ma session ssh.
J’ai pu voir aussi un fichier Excel apparaître dans le répertoire /home/PyGazpar avec le meme nom que ce qu’on peut récupérer sur le site grdf.
Mais à la fin du process, ce fichier disparaît 
Comment faire pour le garder ?

je suis sous Ubuntu 20.10
Merci d’avance 
Cdlt
Philippe

**Comment by ssenart** [comment]
Bonjour,
C'est volontaire d'effacer le fichier à la fin pour éviter de consommer de l'espace inutilement.
Actuellement, il n'y a pas de moyen de le conserver.
Pour quelle raison souhaites-tu le garder ?
Stephane

**Comment by pbranly** [comment]
Merci pour la réponse rapide.
En fait je ne souhaite pas forcément le conserver.
Pour l’instant avec ton appli je vois  les données dans la fenêtre SSH.
Ce que je souhaiterais, c’est de les envoyer dans ma base InfluxDB de façon à pouvoir ensuite en faire des graphiques sous grafana.
Actuellement j’utilise une appli (https://github.com/cyprieng/gazpar-home-assistant ) dans Home Assistant qui alimente sa base Mariadb ( durée une semaine) et la base InfluxDB ( pas de limite pour l’instant)  . Ça fonctionne mais les valeurs sont en vrac. Je trouve toutes les valeurs fournies par grdf, mais HA les affecte à des journées qui ne sont pas les bonnes.
Si tu as une idée, je suis preneur .
merci d’avance 
Philippe 
PS: tu devrais ajouter la ligne « source venv/bin/activate »  dans tes explications d’utilisation .

**Comment by ssenart** [comment]
Je l'utilise aussi avec Home Assistant en utilisant le custom component : https://github.com/ssenart/ha-custom_components/tree/master/gazpar.

Pareillement, j'envoie mes données dans Influxdb pour un affichage avec Grafana.

Mais il y a quelque mois, j'ai découvert le projet https://github.com/kalkih/mini-graph-card pour intégrer des graphes directement dans Lovelace : cela répond à mes besoins de présentation. Du coup, je ne maintiens plus mon lien Influxdb/Grafana.

Concernant les timestamps des events de Home Assistant, je ne connais pas de bonne solution. J'accepte d'avoir un jour de retard (ou plus si GrDF ne fait plus de mise à jour pendant plusieurs jours) entre la donnée et la date associée (affectée par HA). Ces dernières semaines, la publication GrDF est quotidienne, et j'ai bien toutes mes données tous les jours.

PS : Merci pour ta remarque, je vais faire le changement.

**Comment by pbranly** [comment]
J’ai essayé cet après midi ton add-on . J’ai ajouté les fichiers dans customs components mais au reboot de HA, ça m’a fait une erreur ( je ne sais plus … du type impossible de créer le sensor)
J’ai du relancer HA à la hache car le reboot refusait pour une erreur de config 
Je suis en Docker ( pas en supervised)
J’ignore comment intégrer ton add-on et faire le lien avec le python .
Philippe

**Comment by pbranly** [comment]
en fait quel est l’intérêt du format "dictionaries" ? je peux generer un fichier en ajoutant "> file.txt" à la fin de la ligne de commande mais je ne sais pas quoi en faire. tu en fais quoi de ton coté ? il y a plusieurs sujets qui traitent de l'importation de ce type de format dans influxdb mais ils ne sont pas vraiment "user friendly noob dedicated" !

**Comment by ssenart** [comment]
1. C'est vrai que je n'avais donné aucune information comment effectuer l'intégration du Gazpar custom component dans HA. Je viens d'y remédier : https://github.com/ssenart/ha-custom_components
2. PyGazpar est avant tout une librairie Python. Elle est faite pour être intégrer dans un projet Python. La ligne de commande est juste pour illustrer et tester son fonctionnement.
3. Pour l'intégration des données dans InfluxDB en utilisant PyGazpar, je te conseille de partir sur la solution HA et Gazpar custom component comme documenté ici : https://github.com/ssenart/ha-custom_components. Je l'ai déjà fait et cela fonctionne très bien. Je pourrais éventuellement t'aider si tu rencontres des problèmes.

**Comment by pbranly** [comment]
Merci
Première question avant d’y aller 
Puis je l’ntégrer dans un nom de  répertoire différent de gazpar ( déjà utilisé par ailleurs ?


Envoyé depuis mon IPhone SE 2020

**Comment by ssenart** [comment]
_"Puis je l’ntégrer dans un nom de  répertoire différent de gazpar ( déjà utilisé par ailleurs ?"_

=> Je viens de tester et cela fonctionne. Tu peux le faire.

Tiens-moi au courant de ta progression.

**Comment by pbranly** [comment]
J'ai eu des erreurs au debut a cause d'un confliut du a un autre "gazpar"; je l'ai supprimé mais j'ai maintenant :

2021-04-15 13:42:26 ERROR (SyncWorker_5) [custom_components.gazpar.sensor] Failed to query PyGazpar library with exception : Traceback (most recent call last):
File "/usr/local/lib/python3.8/site-packages/selenium/webdriver/common/service.py", line 72, in start
self.process = subprocess.Popen(cmd, env=self.env,
File "/usr/local/lib/python3.8/subprocess.py", line 854, in __init__
self._execute_child(args, executable, preexec_fn, close_fds,
File "/usr/local/lib/python3.8/subprocess.py", line 1702, in _execute_child
raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/usr/local/bin/geckodriver'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
File "/config/custom_components/gazpar/sensor.py", line 124, in update_gazpar_data
client.update()
File "/usr/local/lib/python3.8/site-packages/pygazpar/client.py", line 111, in update
driver = WebDriverWrapper(self.__firefox_webdriver_executable, self.__wait_time, self.__tmp_directory)
File "/usr/local/lib/python3.8/site-packages/pygazpar/webdriverwrapper.py", line 34, in __init__
self.__driver = webdriver.Firefox(executable_path=self.__firefox_webdriver_executable, firefox_profile=profile, options=options, service_log_path=geckodriverLogFile)
File "/usr/local/lib/python3.8/site-packages/selenium/webdriver/firefox/webdriver.py", line 164, in __init__
self.service.start()
File "/usr/local/lib/python3.8/site-packages/selenium/webdriver/common/service.py", line 81, in start
raise WebDriverException(
selenium.common.exceptions.WebDriverException: Message: 'geckodriver' executable needs to be in PATH. 

Le probleme doit etre que mon home assistant dans son docker ne sait pas aller dans /usr/bin/geckodriver
par ailleurs dans le repertoire "/usr/local/lib/python3.8/" je n'ai pas "site-packages" mais j'ai "dist-packages"

**Comment by ssenart** [comment]
Je vois : "FileNotFoundError: [Errno 2] No such file or directory: '/usr/local/bin/geckodriver'"

As-tu bien suivi les étapes décrites ici : https://github.com/ssenart/ha-custom_components ?

Et notamment, mettre le chemin de ton binaires geckodriver dans la config de HA.

Ici, il a l'air de chercher dans /usr/local/bin/geckodriver au lieu de /config/drivers/geckodriver.

**Comment by pbranly** [comment]
bon j'ai remis la bonne ligne pour geckodriver et chargé le fichier dedans.
ca n'a pas voulu au depart pour une question de droits (0644)
je suis passé en 0744 et ca progresse car j'ai les sensors maintenant mais ils sont vides 
j'ai mis des traces detaillées qui donnent:

un detail à corriger plus tard:
2021-04-15 14:16:03 WARNING (MainThread) [homeassistant.loader] No 'version' key in the manifest file for custom integration 'gazpar'. As of Home Assistant 2021.6, this integration will no longer be loaded. Please report this to the maintainer of 'gazpar' 

2021-04-15 14:16:15 DEBUG (SyncWorker_0) [custom_components.gazpar.sensor] Gazpar platform initialization has completed successfully
2021-04-15 14:16:15 DEBUG (SyncWorker_1) [custom_components.gazpar.sensor] HA requests its data to be updated...
2021-04-15 14:16:15 DEBUG (SyncWorker_1) [custom_components.gazpar.sensor] No data available yet for update
2021-04-15 14:16:15 DEBUG (SyncWorker_2) [custom_components.gazpar.sensor] HA requests its data to be updated...
2021-04-15 14:16:15 DEBUG (SyncWorker_2) [custom_components.gazpar.sensor] No data available yet for update
2021-04-15 14:16:15 DEBUG (SyncWorker_1) [custom_components.gazpar.sensor] HA requests its data to be updated...
2021-04-15 14:16:15 DEBUG (SyncWorker_1) [custom_components.gazpar.sensor] No data available yet for update
2021-04-15 14:16:16 DEBUG (SyncWorker_0) [custom_components.gazpar.sensor] HA requests its data to be updated...
2021-04-15 14:16:16 DEBUG (SyncWorker_0) [custom_components.gazpar.sensor] No data available yet for update
2021-04-15 14:16:16 DEBUG (SyncWorker_4) [custom_components.gazpar.sensor] HA requests its data to be updated...
2021-04-15 14:16:16 DEBUG (SyncWorker_4) [custom_components.gazpar.sensor] No data available yet for update

2021-04-15 14:16:17 DEBUG (SyncWorker_3) [custom_components.gazpar.sensor] HA requests its data to be updated...
2021-04-15 14:16:17 DEBUG (SyncWorker_3) [custom_components.gazpar.sensor] No data available yet for update
2021-04-15 14:16:17 DEBUG (SyncWorker_3) [custom_components.gazpar.sensor] HA requests its data to be updated...
2021-04-15 14:16:17 DEBUG (SyncWorker_3) [custom_components.gazpar.sensor] No data available yet for update
2021-04-15 14:16:17 DEBUG (SyncWorker_4) [custom_components.gazpar.sensor] HA requests its data to be updated...
2021-04-15 14:16:17 DEBUG (SyncWorker_4) [custom_components.gazpar.sensor] No data available yet for update
2021-04-15 14:16:18 DEBUG (SyncWorker_8) [custom_components.gazpar.sensor] Querying PyGazpar library for new data...
2021-04-15 14:16:18 ERROR (SyncWorker_8) [custom_components.gazpar.sensor] Failed to query PyGazpar library with exception : Traceback (most recent call last):
  File "/config/custom_components/gazpar/sensor.py", line 124, in update_gazpar_data
    client.update()
  File "/usr/local/lib/python3.8/site-packages/pygazpar/client.py", line 111, in update
    driver = WebDriverWrapper(self.__firefox_webdriver_executable, self.__wait_time, self.__tmp_directory)
  File "/usr/local/lib/python3.8/site-packages/pygazpar/webdriverwrapper.py", line 34, in __init__
    self.__driver = webdriver.Firefox(executable_path=self.__firefox_webdriver_executable, firefox_profile=profile, options=options, service_log_path=geckodriverLogFile)
  File "/usr/local/lib/python3.8/site-packages/selenium/webdriver/firefox/webdriver.py", line 170, in __init__
    RemoteWebDriver.__init__(
  File "/usr/local/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py", line 157, in __init__
    self.start_session(capabilities, browser_profile)
  File "/usr/local/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py", line 252, in start_session
    response = self.execute(Command.NEW_SESSION, parameters)
  File "/usr/local/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/usr/local/lib/python3.8/site-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.SessionNotCreatedException: Message: Expected browser binary location, but unable to find binary in default location, no 'moz:firefoxOptions.binary' capability provided, and no binary flag set on the command line

Je n'ai pas de repertoire "/usr/local/lib/python3.8/site-packages" par contre j'ai "/usr/local/lib/python3.8/dist-packages" dans lequel il y a entres autres Selenium avec les fichiers cités en erreur
Et de toutes facons ils sont en 0644 donc j'ai peur que mon HA en docker n'ait pas les droits

**Comment by ssenart** [comment]
_"selenium.common.exceptions.SessionNotCreatedException: Message: Expected browser binary location, but unable to find binary in default location, no 'moz:firefoxOptions.binary' capability provided, and no binary flag set on the command line"_

=> Il faut aussi installer Firefox et attention à bien prendre une version compatible avec ton geckodriver.

**Comment by pbranly** [comment]
en fait il est installé; si j'ouvre une session ssh depuis mon docker tout y est
et en environnement virtuel ca fonctionne 
le problème c'est que tout tourne en docker ici

**Comment by pbranly** [comment]
tu n'aurais pas un whatsapp, telegram ou autre ?

**Comment by ssenart** [comment]
Je suis aussi sous Docker et ce n'est pas un problème.

Ton Firefox doit être installé dans ton container où tourne HA.

Est-ce que la ligne de commande pygazpar focntionne dans ce même container ?

**Comment by pbranly** [comment]
comment installer firefox dans le docker de home assistant ?
si j'ouvre une session ssh depuis ce container avec portainer, je n'ai meme pas la commande apt-get. 
et non la ligne de commande ne fonctionne surement pas depuis ce container;

**Comment by ssenart** [comment]
C'est tout écrit ici : https://github.com/ssenart/ha-custom_components

Dans le paragraphe : Note about using geckodriver in a Docker container.

Et oui, utiliser portainer pour s'y connecter, c'est bon.

Si c'est l'image officielle HA, alors la commande à utiliser est apk (cf le lien ci-dessus).

**Comment by pbranly** [comment]
Je ne suis pas en image HA ; je suis en home assistant docker. je n'ai pas tout ! le reste se fait soit avec d'autres containers soit en natif linux.
donc par exemple la commande /config/drivers/geckodriver --version fonctionne (c'est deja ca!)
mais je ne peux pas faire apk xxxx car la commande n'est pas installée dans ce bash

**Comment by ssenart** [comment]
C'est quoi la distrib Linux de ton container HA ?

**Comment by pbranly** [comment]
pardon apk fonctionne !
ca donne
WARNING: Ignoring APKINDEX.2c4ac24e.tar.gz: No such file or directory
WARNING: Ignoring APKINDEX.40a3604f.tar.gz: No such file or directory

3.12.3
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.12.3
PRETTY_NAME="Alpine Linux v3.12"
HOME_URL="https://alpinelinux.org/"
BUG_REPORT_URL="https://bugs.alpinelinux.org/"

bash-5.0# apk install firefox
ERROR: 'install' is not an apk command. See 'apk --help'.

**Comment by ssenart** [comment]
Ok, donc la commande:

```bash
apk add firefox
```
devrait fonctionner.

**Comment by pbranly** [comment]
oui je viens de trouver.
et .......... ca marche !!!!!!!!!!! j'ai des valeurs dans les sensors

la question que je me pose : cette installation de firefox se conserve apres reboot ?

sinon j'ai 2 erreurs dans le log:
2021-04-15 15:14:08 WARNING (SyncWorker_5) [pygazpar.webdriverwrapper] find_element_by_xpath('//a[@id='_EPcommonPage_WAR_EPportlet_:formBandeauCnil:j_idt12']'): Cookies accept button -> Not found
2021-04-15 15:14:38 WARNING (SyncWorker_5) [pygazpar.webdriverwrapper] find_element_by_xpath('/html/body/abtasty-modal/div/div[1]'): Advertisement close button -> Not found

**Comment by ssenart** [comment]
- Oui, ça reste après reboot, mais pas après mis à jour de ton container avec la dernière version de ton image: il faudra y penser.
- Ce sont des WARNING, rien de grave.

Donc, prochaine étape, envoyer ça dans InfluxDB avec HA et récupérer ça dans Grafana.

**Comment by pbranly** [comment]
alors il faudra que j'y pense car je suis en version DEV et ca se met a jour tous les jours ! j'aime le risque......
peut etre qu'il est possible d'automatiser la ligne "apk add firefox" quelque part.
Sinon effectivement Influxdb a commencé a recuperer les valeurs 

ce qui est dommage , c'est de ne pas pouvoir uploader tout le fichier de grdf dans influx en une fois. j'aurais l'historique alors que la je repars de aujourdhui

**Comment by ssenart** [comment]
- Oui, c'est possible en fabriquant ta propre image HA à partir de celle que tu utilises, et qui rajoute cette commande : il faut faire un dockerfile et le publier sur dockerhub.
- Pour uploader l'historique dans InfluxDB, regarde de ce côté là : https://docs.influxdata.com/influxdb/v2.0/write-data/developer-tools/csv/

**Comment by pbranly** [comment]
heuu "Oui, c'est possible en fabriquant ta propre image HA à partir de celle que tu utilises, et qui rajoute cette commande : il faut faire un dockerfile et le publier sur dockerhub." j'apprends tous les jours mais la .... 
ca veut dire qu'avec cela, je pourrais continuer a recevoir des upgrades de HA sans casser linstallation de firefox ?

**Comment by ssenart** [comment]
Comment fais-tu les upgrades HA ?

**Comment by pbranly** [comment]
en auto avec watchtower ( je sais ....) 
j'ai essayé d'ajouter command: apk add firefox  dans le docker-compose. le log montre que ca installe et reboote en boucle et j'ai beau fouiller le net, je ne trouve pas comment automatiser l'installation pendant la construction du docker

**Comment by ssenart** [comment]
C'est une bonne idée.

Ce post a l'air assez proche de ce que tu veux faire : https://stackoverflow.com/questions/47615751/docker-compose-run-a-script-after-container-has-started

**Comment by pbranly** [comment]
pas évident. de toutes façons, apres l'instal de firefox, il faut relancer HA afin que ton système fonctionne.
Il faudrait donc pouvoir:
-Au redémarrage quelconque de HA: tester la présence de firefox; si oui c'est tout bon
- Si non, installer Firefox et relancer HA

je suis encore célibataire demain ....... je vais creuser 
merci pour ta patience en tous cas 
.
ce serait bien que tu transformes fil en discussion et non en bug

**Reply by ssenart** [reply]
J'ai modifié mon docker-compose.yml comme suit:

```yaml
version: '2'

services:
  homeassistant:
    image: homeassistant/home-assistant:latest
    container_name: homeassistant
    restart: always
    network_mode: host
    volumes:
      - /share/Container-data/home-assistant/config:/config
    entrypoint: /bin/bash -c "apk add firefox && python -m homeassistant --config /config"
```

...et ça marche nickel !

**Comment by pbranly** [comment]
Fantastique
j'ai relancé HA pas de probleme
ju'ai reconstruit HA, pas de probleme
on verra demain avec un vrai nouveau build
Pour ma culture perso, le "&& python -m homeassistant --config /config sert à quoi ?

**Reply by ssenart** [reply]
Le commande pour lancer HA est :

```bash
python -m homeassistant --config /config
```

La propriété entrypoint du docker-compose se substitue au EntryPoint du Dockerfile.

Du coup, on doit faire notre installation Firefox, ET lancer HA dans le entrypoint du docker-compose.

Le && permet de faire les 2 commandes en séquence, et la seconde sera faite seulement si la première réussit.

**Reply by pbranly** [reply]
Ok compris . J’ai lu que mettre une commande « apk add » fonctionne uniquement pendant la durée de la commande et arrête tout. Ça doit expliquer pourquoi quand je faisais ça, ça relançait sans arrêt le Docker qui lançait la commande qui arrêtait …….
Pffff on en découvre ! Intéressant en tous cas .
Merci monsieur le professeur 😀

**Comment by pbranly** [comment]
Fonctionnement parfait découvert ce matin après mise a jour de HA en version 2021.5.0.dev20210416.

Par contre finalement la mise a jour et l’utilisation de la base Influx est inutile car conduit a des résultats faux.
En effet, HA est fait pour gérer des états et des modifications d’états mais pas une base de données.
Ainsi les valeurs importées sont "les dernières du moment taguées du moment" même si les valeurs sont obsolètes.
Par exemple ce matin ton app a recuperé les 2 dernières valeurs possibles du 12 et13 Avril alors qu'on est le 16. ce sont ces valeurs qui sont ensuite mises a jour dans Influx sans considération de date. (voir extrait de traces fait ce matin dans HA en fin de texte)

Conséquences: impossible de calculer de bonnes valeurs.
c'est pareil également dans HA qui donne à aujourd’hui des valeurs du 13
une solution utilisée par https://github.com/frtz13/homeassistant_gazpar_cl_sensor consiste a mettre a zéro avant minuit et recommencer à tenter de recuperer des données tard dans la journée pendant un temps limité. 
C'est mieux mais ne fonctionne pas quand GRDF tarde a mettre a jour son tableau et dans ce cas, il manque des valeurs perdues définitivement dans la base. 

La seule bonne solution est d'importer le tableau directement dans Infludb (ou autre database) correctement indexées sur le champs date.
ensuite on peut gérer avec grafana et importer le graphe vers HA.
c'est ce que fais avec Linky et ça fonctionne parfaitement

Je doute d'y arriver car je découvre tout ca (même si je suis pugnace !)
j'ai trouvé pas mal de sujets traitant de script python permettant de porter les tables de dictionnaires python en CSV puis de CSV vers Influxdb

Philippe

2021-04-16 06:51:28 DEBUG (SyncWorker_18) [custom_components.gazpar.sensor] data=[{"date": "12/04/2021","start_index_m3": 730.0,"end_index_m3": 735.0,"volume_m3": 5.56,"energy_kwh": 52.0,"converter_factor": "9,475","local_temperature": "4","type": "MES","timestamp": "2021-04-16T06:50:57.245245"},{"date": "13/04/2021","start_index_m3": 735.0,"end_index_m3": 740.0,"volume_m3": 4.21,"energy_kwh": 39.0,"converter_factor": "9,475","local_temperature": "6","type": "MES","timestamp": "2021-04-16T06:50:57.245245"

**Reply by ssenart** [reply]
Essayer de lutter contre le comportement natif de HA n'est pas une solution.

Je laisse HA remplir InfluxDB  à sa façon, ensuite je joue avec les requêtes dans Grafana pour récupérer seulement les données qui m'intéressent. En l'occurence, c'est la dernière valeur publiée tous les jours, et cela s'écrit dans Grafana:

```sql
SELECT last("value") FROM "kWh" WHERE ("entity_id" = 'gazpar_last_energy') AND $timeFilter GROUP BY time(1d) fill(null)
```

Biensûr, si GrDF omet une publication un jour, alors j'aurais, ce jour là, la valeur de la veille. C'est assez rare.

Une autre solution (que je n'ai pas essayé) serait d'utiliser le champ sensor.gazpar_last_period_start_time et/ou sensor.gazpar_last_period_end_time (le champ date dans PyGazpar) comme axe de temps pour requêter dans Grafana.

**Reply by pbranly** [reply]
Le problème c’est que tous les jours, tu n’as que les valeurs de lavant veille …..et encore çe n’est pas sur . 
comment fais tu pour dire à grafana qu’il doit aller chercher la valeur de lavant veille seulement ?
Car sauf erreur « last value » c’est celle récupérée toutes les heures par ton appli . Donc celle vielle de 2 jours 
Me trompe-je 🤭?

**Reply by ssenart** [reply]
En générale, avant minuit du jour J, nous avons la publication de la consommation J-1.

Donc dans Grafana, "last value" du jour J contient la consommation J-1 (la veille).

Du coup, j'interprète chaque pilier de mon graphe comme étant la consommation du jour d'avant.

**Comment by pbranly** [comment]
En général !
Ce matin ( le 16 ) j’avais comme dernière valeur celle du 13
Donc ça rempli toutes les heures des données vieilles d’au moins 2 voire 3 jours
Au mini tu pourrais rapatrier la date liée à celle de la mesure 
Ça permettrait de poubelliser tout ce qui edt identique et ?? Pourrait permettre de gérer un couple homogène dans influxdb
Je n’ai pas vu que tu récupérais cette date dans ton appli 


Envoyé depuis mon IPhone SE 2020

**Reply by ssenart** [reply]
Ah, je n'ai pas le défaut que tu observes. Aujourd'hui, j'ai bien les mesures du 14  (publiée vers 20h00 hier soir).

La date de la mesure est le champ 'date' dans PyGazpar et traduit en champs sensor.gazpar_last_period_start_time/ sensor.gazpar_last_period_end_time dans HA.

**Comment by pbranly** [comment]
Ma dernière valeur est celle du 13 ! C’est ce que je constate dans les dernières traces HA et quand je tente de les voir sur GRDF .
3 jours ! 

{
"date": "12/04/2021",
"start_index_m3": 730.0,
"end_index_m3": 735.0,
"volume_m3": 5.56,
"energy_kwh": 52.0,
"converter_factor": "9,475",
"local_temperature": "4",
"type": "MES",
"timestamp": "2021-04-16T10:50:54.888204"
},
{
"date": "13/04/2021",
"start_index_m3": 735.0,
"end_index_m3": 740.0,
"volume_m3": 4.21,
"energy_kwh": 39.0,
"converter_factor": "9,475",
"local_temperature": "6",
"type": "MES",
"timestamp": "2021-04-16T10:50:54.888204"

**Comment by pbranly** [comment]
je crois comprendre : GRDF ne me donne plus de fichier detaillé "jour" mais un recap par periode !
avec des moulinettes interminables
je pense qu'en plus le site est en rade
Edit: ca refonctionne !
